platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/FeatureProcessor.java

```diff
@@ -0,0 +1,401 @@
+    }
+  }
+
+  /**
+   * Processes the specified feature for uninstallation if the feature in memory is not uninstalled.
+   *
+   * @param feature the current feature from memory
+   * @param id the feature id
+   * @param state the feature state in memory
+   * @param tasks the task list where to record tasks to be executed
+   * @return <code>true</code> if processed; <code>false</code> otherwise
+   */
+  public void processUninstalledFeature(
+      Feature feature, String id, FeatureState state, TaskList tasks) {
+    if (state != FeatureState.Uninstalled) {
+      tasks.add(Operation.UNINSTALL, id, r -> uninstallFeature(r, feature));
+    }
+  }
+
+  /**
+   * Processes the specified feature for installation if the feature in memory is not uninstalled.
+   *
+   * <p><i>Note:</i> A feature that is started in memory will need to be stopped in order to be in
+   * the same resolved state it was on the original system.
+   *
+   * @param feature the current feature from memory
+   * @param id the feature id
+   * @param state the feature state in memory
+   * @param tasks the task list where to record tasks to be executed
+   */
+  public void processInstalledFeature(
+      Feature feature, String id, FeatureState state, String region, TaskList tasks) {
+    if (state == FeatureState.Uninstalled) {
+      tasks.add(Operation.INSTALL, id, r -> installFeature(r, feature));
+    } else if (state == FeatureState.Started) {
+      tasks.add(Operation.STOP, id, r -> stopFeature(r, feature, region));
+    }
+  }
+
+  /**
+   * Processes the specified feature for resolution. If the feature was uninstalled, it will be
+   * installed and then later stopped to get to the resolved state. Since we only want to deal with
+   * one state change at a time, we will purposely increase the number of attempts left for stopping
+   * such that another processing round can be done at which point the state of the feature in
+   * memory will be seen as installed instead of uninstalled like it is right now such that it can
+   * then be finally stopped to be moved into the resolved state as it was on the original system.
+   * Otherwise, if it is not in the resolved state, it will simply be stopped.
+   *
+   * @param feature the current feature from memory
+   * @param id the feature id
+   * @param state the feature state in memory
+   * @param region the region for the feature
+   * @param tasks the task list where to record tasks to be executed
+   */
+  public void processResolvedFeature(
+      Feature feature, String id, FeatureState state, String region, TaskList tasks) {
+    if (state == FeatureState.Uninstalled) {
+      // we need to first install it and on the next round, stop it
+      // as such, let's make sure to increase the number of attempts left for the stop operation
+      tasks.add(Operation.INSTALL, id, r -> installFeature(r, feature));
+      tasks.increaseAttemptsFor(Operation.STOP);
```

Comment:
I am removing the `increaseAttemptsFor()`.

That being said, we couldn't simply add another task here since it would be for nothing since the task list is designed to only execute the tasks for the highest operation and clear the rest. We really need the next iteration that we will re-check everything and re-process the tasks to execute to register that second task.